docs/sts/opa.md

```diff
@@ -0,0 +1,85 @@
+OPA is a lightweight general-purpose policy engine that can be co-located with your service, in this document we talk about how to use OPA HTTP API to authorize Minio STS credentials.
+
+## Get started
+### 1. Prerequisites
+- Docker 18.03 or above, refer here for [installation](https://docs.docker.com/install/).
+- Docker compose 1.20 or above, refere here for [installation](https://docs.docker.com/compose/install/#prerequisites).
+
+### 2. Start OPA
+First, create a `docker-compose.yml` file that runs OPA and the demo web server.
+```
+cat >docker-compose.yml <<EOF
+version: '2'
+services:
+  opa:
+    image: openpolicyagent/opa:0.9.1
+    ports:
+      - 8181:8181
+    command:
+      - "run"
+      - "--server"
+      - "--log-level=debug"
+  api_server:
+    image: openpolicyagent/demo-restful-api:0.2
+    ports:
+      - 5000:5000
+    environment:
+      - OPA_ADDR=http://opa:8181
+      - POLICY_PATH=/v1/data/httpapi/authz
+EOF
+```
+
+Then run `docker-compose` to pull and run the containers.
+```
+docker-compose -f docker-compose.yml up
+```
+
+### 3. Create new OPA Policy
+In another terminal, create a policy that allows users to upload objects
+```
+cat > putobject.rego <<EOF
+package httpapi.authz
+
+import input as http_api
+
+allow {
+ input.action = "s3:PutObject"
+ input.owner = false
+}
+
+EOF
+```
+
+Then load the policy via OPA's REST API.
+```
+curl -X PUT --data-binary @putobject.rego \
+  localhost:8181/v1/policies/putobject
+```
+
+### 4. Setup Minio with OPA
+Minio server expects environment variable for OPA http API url as `MINIO_IAM_OPA_URL`, this environment variable takes a single entry.
+```
+export MINIO_IAM_OPA_URL=http://localhost:8181/v1/data/httpapi/authz
+minio server /mnt/data
+```
+
+### 5. Test with Minio STS API
+Assuming that you have configured Minio server to support STS API by following the doc [Minio STS Quickstart Guide](https://docs.minio.io/docs/minio-sts-quickstart-guide) and once you have obtained the JWT from WSO2 as mentioned in [WSO2 Quickstart Guide](https://docs.minio.io/docs/wso2-quickstart-guide).
+```
+go run sts-example.go -ep http://localhost:9000 -t "eyJ4NXQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJraWQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJhbGciOiJSUzI1NiJ9.eyJhdWQiOiJQb0VnWFA2dVZPNDVJc0VOUm5nRFhqNUF1NVlhIiwiYXpwIjoiUG9FZ1hQNnVWTzQ1SXNFTlJuZ0RYajVBdTVZYSIsImlzcyI6Imh0dHBzOlwvXC9sb2NhbGhvc3Q6OTQ0M1wvb2F1dGgyXC90b2tlbiIsImV4cCI6MTUzNDg5MTc3OCwiaWF0IjoxNTM0ODg4MTc4LCJqdGkiOiIxODQ0MzI5Yy1kNjVhLTQ4YTMtODIyOC05ZGY3M2ZlODNkNTYifQ.ELZ8ujk2Xp9xTGgMqnCa5ehuimaAPXWlSCW5QeBbTJIT4M5OB_2XEVIV6p89kftjUdKu50oiYe4SbfrxmLm6NGSGd2qxkjzJK3SRKqsrmVWEn19juj8fz1neKtUdXVHuSZu6ws_bMDy4f_9hN2Jv9dFnkoyeNT54r4jSTJ4A2FzN2rkiURheVVsc8qlm8O7g64Az-5h4UGryyXU4zsnjDCBKYk9jdbEpcUskrFMYhuUlj1RWSASiGhHHHDU5dTRqHkVLIItfG48k_fb-ehU60T7EFWH1JBdNjOxM9oN_yb0hGwOjLUyCUJO_Y7xcd5F4dZzrBg8LffFmvJ09wzHNtQ"
+
```

Comment:
this example needs to be updated - no sts-example.go in the repo

Reply:
This is fixed